packages/sdk/contractkit/src/kit.test.ts

```diff
@@ -204,3 +205,86 @@ testWithGanache('Fetch whitelisted fee currencies', (web3: Web3) => {
     })
   })
 })
+
+testWithAnvil('kit', (web3) => {
+  let kit: ContractKit
+  let feeToken: StrongAddress
+  beforeAll(async () => {
+    // when this is beforeEach the web3 instance is reused so the second time it already is attached to a celo provider
+    // but if the provider is already an instance of celo provider then  rpccaller is not attached to connection class instance
+    //  web3 must be a new instance when creating a kit not a an instance that was already used by a different kit
+    kit = newKitFromWeb3(web3)
+
+    const feeCurrencyWhitelist = await kit.contracts.getFeeCurrencyWhitelist()
+    const gasOptions = await feeCurrencyWhitelist.getWhitelist()
+    feeToken = gasOptions[0]
+  })
+  describe('populateMaxFeeInToken', () => {
+    describe('when not on cel2', () => {
+      it('throws not L2 error', () => {
+        expect(
+          kit.populateMaxFeeInToken({ feeCurrency: feeToken, gas: '10000000034230982772378193726' })
+        ).rejects.toMatchInlineSnapshot(
+          `[Error: Can't populate \`maxFeeInFeeCurrency\` if not on a CEL2 network]`
+        )
+      })
+    })
+    describe('when on cel2', () => {
+      beforeEach(() => {
+        // replace when setupL2 is in dev tils instead of in cli utils
+        jest.spyOn(web3.eth, 'getCode').mockImplementation(async () => {
+          return '0x2k2131325123PROXY'
+        })
+      })
+      describe('when gas is missing', () => {
+        it('fills the gas and works as normal', () => {
+          expect(
+            kit.populateMaxFeeInToken({
+              feeCurrency: feeToken,
+            })
+          ).resolves.toMatchInlineSnapshot(`
+            {
+              "feeCurrency": "0x2A3733dBc31980f02b12135C809b5da33BF3a1e9",
+              "gas": 53001,
+              "maxFeeInFeeCurrency": "108122040000000",
```

Comment:
so basically that's 102% of max fee (in fee currency) for the transaction? exchange rate for this token in the directory is 1:1, right?

Reply:
yes and correct the exchange rate in contract on anvil seems to always give 1;1.